ndk/src/native_window.rs

```diff
@@ -409,39 +386,32 @@ bitflags::bitflags! {
     /// of those basic transforms.
     #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
     #[doc(alias = "ANativeWindowTransform")]
-    pub struct NativeWindowTransform : u32 {
+    pub struct NativeWindowTransform : i32 {
         #[doc(alias = "ANATIVEWINDOW_TRANSFORM_IDENTITY")]
-        const TRANSFORM_IDENTITY = ffi::ANativeWindowTransform::ANATIVEWINDOW_TRANSFORM_IDENTITY.0;
+        const IDENTITY = ffi::ANativeWindowTransform::ANATIVEWINDOW_TRANSFORM_IDENTITY.0 as i32;
         #[doc(alias = "ANATIVEWINDOW_TRANSFORM_MIRROR_HORIZONTAL")]
-        const TRANSFORM_MIRROR_HORIZONTAL = ffi::ANativeWindowTransform::ANATIVEWINDOW_TRANSFORM_MIRROR_HORIZONTAL.0;
+        const MIRROR_HORIZONTAL = ffi::ANativeWindowTransform::ANATIVEWINDOW_TRANSFORM_MIRROR_HORIZONTAL.0 as i32;
         #[doc(alias = "ANATIVEWINDOW_TRANSFORM_MIRROR_VERTICAL")]
-        const TRANSFORM_MIRROR_VERTICAL = ffi::ANativeWindowTransform::ANATIVEWINDOW_TRANSFORM_MIRROR_VERTICAL.0;
+        const MIRROR_VERTICAL = ffi::ANativeWindowTransform::ANATIVEWINDOW_TRANSFORM_MIRROR_VERTICAL.0 as i32;
         #[doc(alias = "ANATIVEWINDOW_TRANSFORM_ROTATE_90")]
-        const TRANSFORM_ROTATE_90 = ffi::ANativeWindowTransform::ANATIVEWINDOW_TRANSFORM_ROTATE_90.0;
+        const ROTATE_90 = ffi::ANativeWindowTransform::ANATIVEWINDOW_TRANSFORM_ROTATE_90.0 as i32;
+        /// Defined as [`Self::MIRROR_HORIZONTAL`] `|` [`Self::MIRROR_VERTICAL`].
         #[doc(alias = "ANATIVEWINDOW_TRANSFORM_ROTATE_180")]
-        const TRANSFORM_ROTATE_180 = ffi::ANativeWindowTransform::ANATIVEWINDOW_TRANSFORM_ROTATE_180.0;
+        const ROTATE_180 = ffi::ANativeWindowTransform::ANATIVEWINDOW_TRANSFORM_ROTATE_180.0 as i32;
+        /// Defined as [`Self::ROTATE_180`] `|` [`Self::ROTATE_90`].
         #[doc(alias = "ANATIVEWINDOW_TRANSFORM_ROTATE_270")]
-        const TRANSFORM_ROTATE_270 = ffi::ANativeWindowTransform::ANATIVEWINDOW_TRANSFORM_ROTATE_270.0;
+        const ROTATE_270 = ffi::ANativeWindowTransform::ANATIVEWINDOW_TRANSFORM_ROTATE_270.0 as i32;
     }
 }
 
-#[cfg(feature = "api-level-28")]
-#[derive(Debug, Error)]
-pub enum GetDataSpaceError {
-    #[error(transparent)]
-    IoError(#[from] io::Error),
-    #[error(transparent)]
-    TryFromPrimitiveError(#[from] TryFromPrimitiveError<DataSpace>),
-}
-
 /// Compatibility value for [`NativeWindow::set_frame_rate()`]
 #[cfg_attr(
     feature = "api-level-31",
     doc = " and [`NativeWindow::set_frame_rate_with_change_strategy()`]"
 )]
 /// .
 #[cfg(feature = "api-level-30")]
-#[repr(u32)]
+#[repr(i8)]
```

Comment:
Not sure why this is `i8` instead if `i32` - By default I'd generally expect most NDK enums should effectively be represented with the equivalent of a JNI `jint`, since most of them are effectively defined in the Android Java SDK

aah OK, I see this particular enum is passed as an argument to an API that takes an `int8_t` which implies they can't really extend beyond `i8`

Reply:
Ah yeah, that's documented in https://github.com/rust-mobile/ndk/pull/459/commits/da1a7b03f2631f5858e51f983018c2f16d1b1527.

I've been having some chats with the Android folks about fixing their type annotations. Since using clang they're looking to switch to an `enum <name>: <type>` representation and replace e.g. `i8` types in function signatures with `enum FrameRateStrategy` etc.